### PR TITLE
mdx-js v2 update

### LINKED ISSUE #465: Support last version of MDX

### PR DESCRIPTION
**Pull request by PabloSzx**
~~currently getting this error: https://github.com/syntax-tree/unist/discussions/63~~

based on https://github.com/hashicorp/next-mdx-remote/pull/211

I managed to fix all the errors except remark-admonitions https://github.com/elviswolcott/remark-admonitions, which stopped being maintained long ago, but graphql-codegen uses it a lot, that should be the only breaking thing by now (other than now requiring Next.js v12)

closes #465

### CODE REVIEW

**Comment by PabloSzx** [comment]
@B2o5T I managed to fix most of the errors except remark-admonitions, which stopped being maintained long ago, but graphql-codegen uses it a lot https://github.com/elviswolcott/remark-admonitions, that should be the only breaking thing by now (other than requiring Next.js v12) blocking

File: packages/client/src/NextNProgress.tsx [hunk 1]
```diff
@@ -1,7 +1,7 @@
 /* eslint-disable @typescript-eslint/no-non-null-assertion */
 
 import { css, Global } from '@emotion/react';
-import RouterImport from 'next/router';
+import RouterImport from 'next/router.js';
```
Comment (B2o5T) [inline packages/client/src/NextNProgress.tsx:4]: why it's necessary? and other cases with `index.js`

File: packages/mdx-remote/src/format-mdx-error.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+ * Only used if the error object doesn't contain
+ */
+function parsePositionInformationFromErrorMessage(message: string): { start: { line: number; column: number } } | undefined {
+  const positionInfoPattern = /\d+:\d+(-\d+:\d+)/g;
+
+  const match = message.match(positionInfoPattern);
+
+  if (match) {
+    // take the last match, that seems to be the most reliable source of the error.
+    const lastMatch = match.slice(-1)[0];
```
Comment (B2o5T) [inline packages/mdx-remote/src/format-mdx-error.ts:14]:
```suggestion
    const lastMatch = match[match.length - 1];
```

File: packages/client/src/NextNProgress.tsx (hunk 1 above)
Comment (PabloSzx) [reply packages/client/src/NextNProgress.tsx:4]: this is part of esm, it's not new

Comment (PabloSzx) [reply packages/client/src/NextNProgress.tsx:4]: Next.js doesn't have explicit package exports, which means that deep imports need explicit extensions, it's just how it works

**Comment by charlypoly** [comment]
@B2o5T do we plan to merge this soon?

The migration of the codegen doc can be quite easy by replacing `remark-admonitions` with custom components 🤓

**Pull request merged**